### PR TITLE
Add matrix norms

### LINKED ISSUE #44: Add matrix_frob_norm, matrix_one_norm, and matrix_inf_norm
Add `matrix_frob_norm`, `matrix_one_norm`, and `matrix_inf_norm` (thanks to coauthor Piotr Luszczek).

### PR DESCRIPTION
**Pull request by amklinv-nnl**
Closes #44.

### CODE REVIEW

File: include/experimental/__p1673_bits/blas1_matrix_frob_norm.hpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  auto result = init;
+  if (A.extent(0) == 0 || A.extent(1) == 0) {
+    return result;
+  }
+  else if(A.extent(0) == ptrdiff_t(1) && A.extent(1) == ptrdiff_t(1)) {
+    result += abs(A(0, 0));
+    return result;
+  }
+
+  // Rescaling avoids unwarranted overflow or underflow.
```
Comment (mhoemmen) [inline include/experimental/__p1673_bits/blas1_matrix_frob_norm.hpp:72]: Nicely done! : - )  At some point (not now), we could consider rewriting this using `vector_sum_of_squares` (see `[linalg.algs.blas1.ssq]`), or factoring out a generic algorithm for this purpose.

File: include/experimental/__p1673_bits/blas1_matrix_frob_norm.hpp [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+         class in_matrix_t,
+         class Scalar>
+Scalar matrix_frob_norm(ExecutionPolicy&& exec,
+                        in_matrix_t A,
+                        Scalar init)
+{
+  return matrix_frob_norm(A, init);
+}
+
+// TODO: Implement auto functions
```
Comment (mhoemmen) [inline include/experimental/__p1673_bits/blas1_matrix_frob_norm.hpp:104]: The proposal explains that the return type is `decltype(abs(A(0,0)) * abs(A(0,0)))`.  The resulting function would look like this:
```
template<class in_matrix_t>
auto matrix_frob_norm(in_matrix_t A) -> decltype(abs(A(0,0)) * abs(A(0,0)))
{
  using T = decltype(abs(A(0,0)) * abs(A(0,0)));
  return matrix_frob_norm(A, T{});
}
```

File: include/experimental/__p1673_bits/blas1_matrix_one_norm.hpp [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  auto result = init;
+  if (A.extent(0) == 0 || A.extent(1) == 0) {
+    return result;
+  }
+  else if(A.extent(0) == ptrdiff_t(1) && A.extent(1) == ptrdiff_t(1)) {
+    result += abs(A(0, 0));
+    return result;
+  }
+
+  // These loops can be rearranged for optimal memory access patterns,
```
Comment (mhoemmen) [inline include/experimental/__p1673_bits/blas1_matrix_one_norm.hpp:72]: I'm not quite sure what you mean by this comment.  The algorithm only reads each entry of the matrix once, so it shouldn't help much to use any more than O(1) storage.

**Review by mhoemmen — APPROVED** [review]
Thanks @amklinv-nnl ! : - D  We can leave the TODOs for later.

File: include/experimental/__p1673_bits/blas1_matrix_inf_norm.hpp [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+#include <cstdlib>
+
+namespace std {
+namespace experimental {
+inline namespace __p1673_version_0 {
+namespace linalg {
+
+template<class in_matrix_t,
+         class Scalar>
+Scalar matrix_one_norm(in_matrix_t A,
```
Comment (amklinv-nnl) [inline include/experimental/__p1673_bits/blas1_matrix_inf_norm.hpp:56]: Whoops.

File: include/experimental/__p1673_bits/blas1_matrix_one_norm.hpp (hunk 3 above)
Comment (mhoemmen) [reply include/experimental/__p1673_bits/blas1_matrix_one_norm.hpp:72]: Resolved out of band; thanks! : - )

**Pull request merged**